### PR TITLE
Fix building statically without any dso support

### PR DESCRIPTION
**Pull request by bernd-edlinger**
I want to be able to build

./Configure linux-generic32 no-shared no-dso -DDSO_NONE

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
Note, this seems to be broken, no-dso does not make the test skipped:
https://github.com/openssl/openssl/blob/3c90534830fdb0cc0e790dd1d5fa2a90e1375e87/test/recipes/90-test_shlibload.t#L22-L24

File: crypto/include/internal/dso_conf.h.in [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 #ifndef HEADER_DSO_CONF_H
 # define HEADER_DSO_CONF_H
+# ifndef DSO_NONE
```
Comment (levitte) [inline crypto/include/internal/dso_conf.h.in]: catch 22: `DSO_NONE` is defined by the perl code below, if `$target{dso_scheme}` is undefined.

This isn't the right way to do this.  If you truly want this, you must make `dso` disablable again, in `Configure`.  You do so by adding it to `@disablables` and removing it from `%deprecated_disablables`, and then you need to change this script as follows:

``` diff
-    my $scheme = uc $target{dso_scheme};
+    my $scheme = $disabled{dso} ? undef : uc $target{dso_scheme};
```

Comment (bernd-edlinger) [reply crypto/include/internal/dso_conf.h.in]: That would be compelling, but when I do that the while crypto/dso directory
is no longer built, but I need the stubs in dso_lib.c, which just return not-implemented.

Comment (levitte) [reply crypto/include/internal/dso_conf.h.in]: This fixes that:

```
diff --git a/Configure b/Configure
index 5a699836f3..cf0ee02e85 100755
--- a/Configure
+++ b/Configure
@@ -1181,7 +1181,7 @@ foreach my $what (sort keys %disabled) {
         my $macro = $disabled_info{$what}->{macro} = "OPENSSL_NO_$WHAT";
 
         if ((grep { $what eq $_ } @{$config{sdirs}})
-                && $what ne 'async' && $what ne 'err') {
+                && $what ne 'async' && $what ne 'err' && $what ne 'dso') {
             @{$config{sdirs}} = grep { $what ne $_} @{$config{sdirs}};
             $disabled_info{$what}->{skipped} = [ catdir('crypto', $what) ];
 
```

**Comment by levitte** [comment]
Something I forgot is that documentation should be added in `INSTALL`.  You can look in 1.1.0 to see how it was previously documented.

This goes for #9895 as well, of course...

**Comment by bernd-edlinger** [comment]
Merged to 1.1.1 as 8dcd574. Thanks!

**Pull request closed without merging**